test/Registry.t.sol

```diff
@@ -22,7 +22,7 @@ contract RegistryTest is Test {
     address private USDC_USD_FEED = 0x8fFfFfd4AfB6115b954Bd326cbe7B4BA576818f6;
 
     function setUp() external {
-        priceRouter = new PriceRouter();
+        priceRouter = new PriceRouter(registry);
         registry = new Registry(gravityBridge, swapRouter, address(priceRouter));
```

Comment:
Another design concept to confirm:
This `PriceRouterV2` will be deployed and ideally replace the current `SLOT` for the `PriceRouterV1`, thereby updating Cellars querying for price, right? 

How does that transition look? Is it something along the lines of: 

-  PriceRouterV2 approved via audit, finalization, and governance.
- `Registry` simply updates the `PRICE_ROUTER_REGISTRY_SLOT` with new `PriceRouterV2`?

Is there an iterative approach to updating things?

Reply:
Yesm this process, then RYE, and RYGOV cellars will need to call `cachePriceRouter` to use the newest one. Though they dont need to do that until they need to start pricing some exotic asset like balancer lp

Reply:
We discussed this, and the opt-in required from the Cellars makes sense with this process.